plugins/org.python.pydev/tests_completions/org/python/pydev/ast/codecompletion/PythonCompletionWithBuiltinsPython3Test.java

```diff
@@ -297,4 +299,48 @@ public void testCodeCompletionPep484Return2() throws Exception {
         requestCompl(s, s.length(), -1, new String[] { "bar()" });
     }
 
+    public void testCompletionWithWalrus() throws Exception {
+        String s;
+        String original = "" +
+                "def test_anything():\n" +
+                "    while a := 10:\n" +
+                "        a.";
+        s = StringUtils.format(original, "");
+        ICompletionProposalHandle[] proposals = requestCompl(s, s.length(), -1,
+                new String[] { "as_integer_ratio()", "bit_length()", "conjugate()", "denominator" });
+        assertEquals(72, proposals.length);
+    }
+
+    public void testCompletionWithWalrus2() throws Exception {
+        String s;
+        String original = "" +
+                "while (x := 10) > 5:\n" +
+                "  x.";
+        s = StringUtils.format(original, "");
+        ICompletionProposalHandle[] proposals = requestCompl(s, s.length(), -1,
+                new String[] { "as_integer_ratio()", "bit_length()", "conjugate()", "denominator" });
+        assertEquals(72, proposals.length);
+    }
+
+    public void testCompletionWithWalrus3() throws Exception {
+        String s;
+        String original = "" +
+                "if x := 10:\n" +
+                "  x.";
+        s = StringUtils.format(original, "");
+        ICompletionProposalHandle[] proposals = requestCompl(s, s.length(), -1,
+                new String[] { "as_integer_ratio()", "bit_length()", "conjugate()", "denominator" });
+        assertEquals(72, proposals.length);
```

Comment:
You should not check how many proposals are there in total (this makes the test brittle as it's very likely it'll change in a new python release).